packages/gatsby/src/utils/start-server.ts

```diff
@@ -278,13 +278,22 @@ module.exports = {
 
       if (page) {
         try {
-          const pageData: IPageDataWithQueryResult = process.env
-            .GATSBY_EXPERIMENTAL_QUERY_ON_DEMAND
-            ? await getPageDataExperimental(page.path)
-            : await readPageData(
-                path.join(store.getState().program.directory, `public`),
-                page.path
-              )
+          let pageData: IPageDataWithQueryResult
+          if (process.env.GATSBY_EXPERIMENTAL_QUERY_ON_DEMAND) {
+            const start = Date.now()
+
+            pageData = await getPageDataExperimental(page.path)
```

Comment:
This will also measure time taken by anything else before this `await` promise resolves. Are you sure this is reliable?

Reply:
I think it correctly measures how long people have to wait. So even if something else affects this promise, it should capture the actual user experience.